src/util/string.cpp

```diff
@@ -889,3 +889,91 @@ std::wstring translate_string(const std::wstring &s)
 	return translate_string(s, g_client_translations);
 #endif
 }
+
+static const std::wstring disallowed_dir_names[] {
```

Comment:
consider making this a `std::array` so the length becomes a compile error if it does not match